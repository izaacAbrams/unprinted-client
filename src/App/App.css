body {
	margin: 0;
	padding: 0;
	box-sizing: border-box;
	background-color: #f4d35e;
}

.red {
	font-size: 16px;
	color: red;
}
