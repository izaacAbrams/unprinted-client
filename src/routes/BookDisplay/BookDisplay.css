.BookDisplay__img {
	height: 400px;
	align-self: center;
}

.BookDisplay__book {
	background-color: #efefe0;
	display: flex;
	flex-direction: column;
	align-items: baseline;
	padding: 5%;
}

.BookDisplay__info {
	text-align: center;
	padding: 15px 30px;
	display: flex;
	flex-direction: column;
}

.BookDisplay__info > h1 {
	font-family: "EB Garamond", serif;
	font-size: 32px;
}

.BookDisplay__info > h2 {
	font-family: "EB Garamond", serif;
	font-size: 18px;
}

.BookDisplay__info > p {
	font-family: "Montserrat", sans-serif;
}

.BookDisplay__btn {
	font-family: "EB Garamond", serif;
	font-size: 16px;
	padding: 5px 20px;
	border-radius: 10px;
	background-color: #1c1f22;
	color: #efefe0;
	text-decoration: none;
	margin-top: 25px;
}

.BookDisplay__btn:hover {
	background-color: #f4d35e;
	color: #1c1f22;
}
