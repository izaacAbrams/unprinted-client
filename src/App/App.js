import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";
import LandingPage from "../routes/LandingPage/LandingPage";
import Nav from "../components/Nav/Nav";
import BooksList from "../routes/BooksList/BooksList";
import NewBook from "../routes/NewBook/NewBook";
import Account from "../routes/Account/Account";
import BookDisplay from "../routes/BookDisplay/BookDisplay";
import UnprintedContext from "../context/UnprintedContext";
import CreatedBooksList from "../routes/CreatedBookList/CreatedBookList";
import EditBook from "../routes/EditBook/EditBook";
import AddChapter from "../routes/AddChapter/AddChapter";
import BookReader from "../routes/BookReader/BookReader";
import LoginPage from "../routes/LoginPage/LoginPage";
import Checkout from "../routes/Checkout/Checkout";
import SignUpForm from "../routes/SignUpForm/SignUpForm";
import BookApiService from "../services/book-api-services";
import PrivateRoute from "../services/PrivateRoute";
import "./App.css";

class App extends Component {
	static contextType = UnprintedContext;

	state = {
		library: [],
		createdLibrary: [],
		ownedLibrary: [],
		updateSignedIn: (status) => {
			this.setState({ isSignedIn: status });
		},
		addCreatedLibrary: (books) => {
			this.setState({ createdLibrary: books });
		},
		addOwnedLibrary: (books) => {
			this.setState({ ownedLibrary: books });
		},
		getCurrentBook: (book_id) => {
			return this.state.library.find((book) => book.id.toString() === book_id);
		},
		getCreatedBook: (book_id) => {
			return this.state.createdLibrary.find(
				(book) => book.id.toString() === book_id
			);
		},
		addBook: (book) => {
			this.state.library.push(book);
		},
		addSection: (newSection, book_id) => {
			const current = this.state.getCurrentBook(book_id);
			const currentIndex = this.state.library.indexOf(current);
			this.state.library[currentIndex].content.push(newSection);
		},
		addImageData: (imageData) => {
			this.setState({
				image_data: imageData,
			});
		},
		editChapter: (book_id, chapter, newChapter) => {
			const current = this.state.getCreatedBook(book_id);
			const currentIndex = this.state.createdLibrary.indexOf(current);
			const chapterIndex = this.state.createdLibrary[
				currentIndex
			].content.findIndex((content) => content.section === parseInt(chapter));
			this.state.createdLibrary[currentIndex].content.splice(chapterIndex, 1);
			this.state.createdLibrary[currentIndex].content.push(newChapter);
			BookApiService.updateBook(current);
		},
		removeBook: (id) => {
			BookApiService.removeBook(id);
			this.setState({
				createdLibrary: this.state.createdLibrary.filter(
					(book) => book.id !== id
				),
			});
		},
	};

	componentDidMount() {
		BookApiService.getLibrary().then((books) =>
			this.setState({ library: books })
		);
		BookApiService.getOwnedLibrary().then((books) =>
			this.state.addOwnedLibrary(books)
		);
	}
	render() {
		const renderLibrary =
			this.state.library.length > 0 ? <></> : <p>Loading</p>;
		return (
			<div className="App">
				<header className="App-header">
					<Nav />
				</header>
				{renderLibrary}
				<UnprintedContext.Provider value={this.state}>
					<Switch>
						<Route exact path={"/"} component={LandingPage} />
						<Route path={"/login"} component={LoginPage} />
						<Route path={"/signup"} component={SignUpForm} />
						<PrivateRoute path={"/new"} component={NewBook} />
						<PrivateRoute path={"/account"} component={Account} />
						<Route path={"/book-list"} component={BooksList} />
						<Route path={"/book/:book_id"} component={BookDisplay} />
						<Route path={"/checkout/:book_id"} component={Checkout} />
						<PrivateRoute path={"/my-books"} component={CreatedBooksList} />
						<PrivateRoute
							path={"/edit/:book_id/:section"}
							component={EditBook}
						/>
						<PrivateRoute
							path={"/:book_id/add-chapter"}
							component={AddChapter}
						/>
						<PrivateRoute path={"/read/:book_id"} component={BookReader} />
					</Switch>
				</UnprintedContext.Provider>
			</div>
		);
	}
}

export default App;
