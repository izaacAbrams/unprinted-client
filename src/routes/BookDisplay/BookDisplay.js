import React, { Component } from "react";
import { Link } from "react-router-dom";
import Checkout from "../Checkout/Checkout";
import UnprintedContext from "../../context/UnprintedContext";
import "./BookDisplay.css";
import TokenService from "../../services/token-service";

class BookDisplay extends Component {
	static contextType = UnprintedContext;

	render() {
		const current = this.context.getCurrentBook(
			this.props.match.params.book_id
		);
		const isOwned = !!this.context.ownedLibrary.find(
			(book) => book.id === current.id
		);

		const current_book = this.context.getCurrentBook(
			this.props.match.params.book_id
		) ? (
			<div className="BookDisplay__book">
				<img
					src={current.cover_img}
					className="BookDisplay__img"
					alt={`Cover for ${current.title}`}
				/>
				<div className="BookDisplay__info">
					<h1>{current.title}</h1>
					<h2>{current.author}</h2>
					<p>{current.summary}</p>
					{isOwned ? (
						<Link to={`/read/${current.id}`}>Read</Link>
					) : (
						<Checkout
							current={this.props.match.params.book_id}
							price={current.price}
						/>
					)}
				</div>
			</div>
		) : (
			<></>
		);
		return <div className="BookDisplay">{current_book}</div>;
	}
}

export default BookDisplay;
