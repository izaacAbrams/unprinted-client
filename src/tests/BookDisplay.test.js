import React from "react";
import ReactDOM from "react-dom";
import BookDisplay from "../routes/BookDisplay/BookDisplay";
import renderer from "react-test-renderer";
import App from "../App/App";
import { BrowserRouter } from "react-router-dom";

describe("bookdisplay component", () => {
	it("renders book display page", () => {
		const div = document.createElement("div");

		ReactDOM.render(
			<BrowserRouter>
				<App>
					<BookDisplay />
				</App>
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});
	it("renders the UI as expected", () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<App>
						<BookDisplay />
					</App>
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
