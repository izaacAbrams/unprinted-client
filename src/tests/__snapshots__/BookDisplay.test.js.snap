// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookdisplay component renders the UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="Nav"
    >
      <div
        className="Nav__logo"
      >
        <a
          className="Nav__link Nav__link_logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Unprinted Logo"
            className="Nav__logo_img"
            src="logo.png"
          />
          Unprinted
        </a>
      </div>
      <a
        className="Nav__link"
        href="/book-list"
        onClick={[Function]}
      >
        Library
      </a>
      <div
        className="Header__not-logged-in"
      >
        <div
          className="Nav__section"
        >
          <a
            className="Nav__register Nav__link"
            href="/signup"
            onClick={[Function]}
          >
            Register
          </a>
          <a
            className="Nav__login Nav__link"
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </div>
      </div>
    </div>
  </header>
  <p>
    Loading
  </p>
  <div
    className="LandingPage"
  >
    <h1
      className="LandingPage__title"
    >
      Unprinted
    </h1>
    <p
      className="LandingPage__p"
    >
      The crowdsourced ebook library. If you are looking for something to read, check out the library! Without signing up/logging in you are able to view general information about the book, such as the cover image, summary, and price. In order to purchase a book and read it, you must first log in and then you will have the option to puchase the book.
      <br />
      <br />
      If you are an author, create an account to start on a new book. Once the initial book information has been made, you can go to the 'my-books' page in your account to add new chapters. 
      <br />
      <br />
       To demo the login, use the email leela@gmail.com and the password leelapassword. To demo purchasing a book, use the card number 4242 4242 4242 4242 with any name/CVC/zip code.
    </p>
    <a
      className="LandingPage__link"
      href="/book-list"
      onClick={[Function]}
    >
      Check out the library
    </a>
  </div>
</div>
`;
